libs/application/templates/marriage-conditions/src/lib/dataSchema.ts

```diff
@@ -40,55 +40,68 @@ export const dataSchema = z.object({
   ceremony: z
     .object({
       hasDate: z.string(),
-      withDate: z
+      date: z.string(),
+      period: z
         .object({
-          date: z.string(),
-          ceremonyPlace: z.string(),
-          office: z.string(),
-          society: z.string(),
-        })
-        .partial()
-        .refine(({ date }) => !!date, {
-          message: coreErrorMessages.defaultError.defaultMessage,
-          path: ['date'],
-        })
-        .refine(({ ceremonyPlace }) => !!ceremonyPlace, {
-          message: coreErrorMessages.defaultError.defaultMessage,
-          path: ['ceremonyPlace'],
+          dateFrom: z.string().optional(),
+          dateTil: z.string().optional(),
```

Comment:
Why not use [`zod.date`](https://github.com/colinhacks/zod#dates) for fields representing dates?